diff --git a/config/BUILDCONFIG.gn b/config/BUILDCONFIG.gn
index 147226d..92ef627 100644
--- a/config/BUILDCONFIG.gn
+++ b/config/BUILDCONFIG.gn
@@ -287,10 +287,7 @@ if (target_os == "android") {
 } else if (target_os == "zos") {
   _default_toolchain = "//build/toolchain/zos:$target_cpu"
 } else if (target_os == "emscripten") {
-  # Because it's too hard to remove all targets from //BUILD.gn that do not work with it.
-  assert(
-      false,
-      "emscripten is not a supported target_os. It is available only as secondary toolchain.")
+  _default_toolchain = "//build/toolchain/wasm:$target_cpu"
 } else {
   assert(false, "Unsupported target_os: $target_os")
 }
diff --git a/config/compiler/BUILD.gn b/config/compiler/BUILD.gn
index 5103543..6bf11a8 100644
--- a/config/compiler/BUILD.gn
+++ b/config/compiler/BUILD.gn
@@ -309,6 +309,8 @@ config("compiler") {
     configs += [ "//build/config/aix:compiler" ]
   } else if (current_os == "zos") {
     configs += [ "//build/config/zos:compiler" ]
+  } else if (current_os == "emscripten") {
+    configs += [ "//build/config/wasm:compiler" ]
   }
 
   configs += [
@@ -395,7 +397,7 @@ config("compiler") {
       } else if (is_chromeos) {
         cflags += [ "-fstack-protector-strong" ]
       } else if (is_posix || is_fuchsia) {
-        if (current_os != "aix") {
+        if (current_os != "aix" && current_os != "emscripten") {
           # Not available on aix.
           cflags += [ "-fstack-protector" ]
         }
diff --git a/toolchain/wasm/BUILD.gn b/toolchain/wasm/BUILD.gn
index 4b4d65c..08612bf 100644
--- a/toolchain/wasm/BUILD.gn
+++ b/toolchain/wasm/BUILD.gn
@@ -25,6 +25,9 @@ gcc_toolchain("wasm") {
   nm = cc
   ld = cxx
 
+  extra_cflags = "-Wno-unknown-warning-option"
+  extra_cxxflags = "-Wno-unknown-warning-option"
+
   toolchain_args = {
     current_cpu = "wasm"
     current_os = "emscripten"
