diff --git a/core/fxcrt/BUILD.gn b/core/fxcrt/BUILD.gn
index 90b9ec7..1e9be68 100644
--- a/core/fxcrt/BUILD.gn
+++ b/core/fxcrt/BUILD.gn
@@ -150,7 +150,7 @@ source_set("fxcrt") {
   } else {
     sources += [ "fx_memory_malloc.cpp" ]
   }
-  if (is_posix) {
+  if (is_posix || is_wasm) {
     sources += [
       "cfx_fileaccess_posix.cpp",
       "cfx_fileaccess_posix.h",
diff --git a/core/fxge/BUILD.gn b/core/fxge/BUILD.gn
index 215a63b..e52e9b6 100644
--- a/core/fxge/BUILD.gn
+++ b/core/fxge/BUILD.gn
@@ -160,7 +160,7 @@ source_set("fxge") {
     ]
   }
 
-  if (is_linux || is_chromeos) {
+  if (is_linux || is_chromeos || is_wasm) {
     sources += [ "linux/fx_linux_impl.cpp" ]
   }
 
