diff --git a/BUILD.gn b/BUILD.gn
index 4edba9f43..8625eb233 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -271,6 +271,34 @@ component("pdfium") {
     complete_static_lib = true
     configs -= [ "//build/config/compiler:thin_archive" ]
   }
+
+  if (is_wasm) {
+    exported_functions = exec_script(
+      "//build/toolchain/wasm/exported_functions.py",
+      [rebase_path("public")],
+      "trim string"
+    )
+
+    ldflags = [
+      "-sEXPORTED_FUNCTIONS=$exported_functions",
+      "-sLLD_REPORT_UNDEFINED",
+      "-sWASM=1",
+      "-sALLOW_MEMORY_GROWTH=1",
+      "-sEXPORTED_RUNTIME_METHODS=ccall,cwrap",
+      "--no-entry",
+    ]
+
+    if (is_debug) {
+      ldflags += [
+        "--profile",
+        "-g",
+      ]
+    } else {
+      ldflags += [
+        "-O2",
+      ]
+    }
+  }
 }
 
 # Targets below this are only visible within this file.
diff --git a/core/fxcrt/BUILD.gn b/core/fxcrt/BUILD.gn
index 21060a47b..0bb732909 100644
--- a/core/fxcrt/BUILD.gn
+++ b/core/fxcrt/BUILD.gn
@@ -158,7 +158,7 @@ source_set("fxcrt") {
   } else {
     sources += [ "fx_memory_malloc.cpp" ]
   }
-  if (is_posix) {
+  if (is_posix || is_wasm) {
     sources += [
       "cfx_fileaccess_posix.cpp",
       "cfx_fileaccess_posix.h",
diff --git a/core/fxge/BUILD.gn b/core/fxge/BUILD.gn
index a07bcd089..21b063664 100644
--- a/core/fxge/BUILD.gn
+++ b/core/fxge/BUILD.gn
@@ -160,7 +160,7 @@ source_set("fxge") {
     ]
   }
 
-  if (is_linux || is_chromeos) {
+  if (is_linux || is_chromeos || is_wasm) {
     sources += [ "linux/fx_linux_impl.cpp" ]
   }
 
