diff --git a/core/fxcrt/BUILD.gn b/core/fxcrt/BUILD.gn
index b92e1d667..b67b57ca9 100644
--- a/core/fxcrt/BUILD.gn
+++ b/core/fxcrt/BUILD.gn
@@ -126,7 +126,7 @@ source_set("fxcrt") {
     "//third_party/icu:icuuc",
   ]
 
-  if (is_posix || is_fuchsia) {
+  if (is_posix || is_fuchsia || is_wasm) {
     sources += [
       "cfx_fileaccess_posix.cpp",
       "cfx_fileaccess_posix.h",
diff --git a/core/fxge/BUILD.gn b/core/fxge/BUILD.gn
index 18b15b70a..cd42a8eaf 100644
--- a/core/fxge/BUILD.gn
+++ b/core/fxge/BUILD.gn
@@ -163,7 +163,7 @@ source_set("fxge") {
     ]
   }
 
-  if (is_linux || is_chromeos || is_fuchsia) {
+  if (is_linux || is_chromeos || is_fuchsia || is_wasm) {
     sources += [ "linux/fx_linux_impl.cpp" ]
   }
 
