diff --git a/config/clang/BUILD.gn b/config/clang/BUILD.gn
index c38670a..5911f4b 100644
--- a/config/clang/BUILD.gn
+++ b/config/clang/BUILD.gn
@@ -183,7 +183,7 @@ template("clang_lib") {
         } else if (current_cpu == "arm64") {
           _suffix = "-aarch64"
         } else {
-          assert(false)  # Unhandled cpu type
+          # Unhandled cpu type
         }
       } else if (is_apple) {
         _dir = "darwin"
@@ -205,7 +205,7 @@ template("clang_lib") {
         } else if (current_cpu == "s390x") {
           _dir = "s390x-unknown-linux-gnu"
         } else {
-          assert(false)  # Unhandled cpu type
+          # Unhandled cpu type
         }
       } else if (is_fuchsia) {
         if (current_cpu == "x64") {
@@ -213,7 +213,7 @@ template("clang_lib") {
         } else if (current_cpu == "arm64") {
           _dir = "aarch64-unknown-fuchsia"
         } else {
-          assert(false)  # Unhandled cpu type
+          # Unhandled cpu type
         }
       } else if (is_android) {
         _dir = "linux"
@@ -228,15 +228,20 @@ template("clang_lib") {
         } else if (current_cpu == "riscv64") {
           _suffix = "-riscv64-android"
         } else {
-          assert(false)  # Unhandled cpu type
+          # Unhandled cpu type
         }
       } else {
-        assert(false)  # Unhandled target platform
+        # Unhandled target platform
       }
 
       _clang_lib_dir = "$clang_base_path/lib/clang/$clang_version/lib"
       _lib_file = "${_prefix}clang_rt.${_libname}${_suffix}.${_ext}"
-      libs = [ "$_clang_lib_dir/$_dir/$_lib_file" ]
+      if (_dir != "") {
+        libs = [ "$_clang_lib_dir/$_dir/$_lib_file" ]
+      } else {
+        print("WARNING: No clang runtime library for $target_os/$current_cpu")
+        print("$_clang_lib_dir/????/$_lib_file")
+      }
     }
   }
 }
