diff --git a/BUILD.gn b/BUILD.gn
index c19e2227a..bbd94ef88 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -23,6 +23,7 @@ config("pdfium_common_config") {
     "OPJ_STATIC",
     "PNG_PREFIX",
     "PNG_USE_READ_MACROS",
+    "FPDFSDK_EXPORTS",
   ]
 
   if (pdf_enable_v8) {
@@ -117,7 +118,7 @@ jumbo_source_set("pdfium_public_headers") {
   ]
 }
 
-jumbo_static_library("pdfium") {
+shared_library("pdfium") {
   sources = [
     "fpdfsdk/fpdf_annot.cpp",
     "fpdfsdk/fpdf_attachment.cpp",
diff --git a/public/fpdfview.h b/public/fpdfview.h
index ac109b9b3..96ef68555 100644
--- a/public/fpdfview.h
+++ b/public/fpdfview.h
@@ -154,13 +154,22 @@ typedef int FPDF_ANNOT_APPEARANCEMODE;
 // Dictionary value types.
 typedef int FPDF_OBJECT_TYPE;
 
-#if defined(_WIN32) && defined(FPDFSDK_EXPORTS)
-// On Windows system, functions are exported in a DLL
-#define FPDF_EXPORT __declspec(dllexport)
-#define FPDF_CALLCONV __stdcall
+#ifdef FPDFSDK_EXPORTS
+# if defined(_WIN32)
+#  define FPDF_EXPORT __declspec(dllexport)
+#  define FPDF_CALLCONV __stdcall
+# else
+#  define FPDF_EXPORT __attribute__((visibility("default")))
+#  define FPDF_CALLCONV
+# endif
 #else
-#define FPDF_EXPORT
-#define FPDF_CALLCONV
+# if defined(_WIN32)
+#  define FPDF_EXPORT __declspec(dllimport)
+#  define FPDF_CALLCONV __stdcall
+# else
+#  define FPDF_EXPORT
+#  define FPDF_CALLCONV
+# endif
 #endif
 
 // Exported Functions
