diff --git a/BUILD.gn b/BUILD.gn
index 62210c2f6..053fbff6d 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -341,7 +341,7 @@ group("pdfium_public_headers") {
   ]
 }
 
-component("pdfium") {
+shared_library("pdfium") {
   output_name = "pdfium"
   libs = []
   configs += [ ":pdfium_strict_config" ]
@@ -381,6 +381,9 @@ component("pdfium") {
       "gdi32.lib",
       "user32.lib",
     ]
+    sources = [
+      "resources.rc"
+    ]
   }
 
   if (is_mac) {
