diff --git a/BUILD.gn b/BUILD.gn
index abdd41456..ec82e8fd8 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -24,6 +24,7 @@ config("pdfium_common_config") {
     "OPJ_STATIC",
     "PNG_PREFIX",
     "PNG_USE_READ_MACROS",
+    "FPDFSDK_EXPORTS"
   ]
 
   if (pdf_enable_v8) {
@@ -101,7 +102,7 @@ config("xfa_warnings") {
   }
 }
 
-jumbo_static_library("pdfium") {
+jumbo_shared_library("pdfium") {
   sources = [
     "fpdfsdk/cba_annotiterator.cpp",
     "fpdfsdk/cba_annotiterator.h",
diff --git a/public/fpdfview.h b/public/fpdfview.h
index ffa467834..e0a400393 100644
--- a/public/fpdfview.h
+++ b/public/fpdfview.h
@@ -151,13 +151,22 @@ typedef int FPDF_ANNOT_APPEARANCEMODE;
 // Dictionary value types.
 typedef int FPDF_OBJECT_TYPE;
 
-#if defined(_WIN32) && defined(FPDFSDK_EXPORTS)
-// On Windows system, functions are exported in a DLL
-#define FPDF_EXPORT __declspec(dllexport)
-#define FPDF_CALLCONV __stdcall
+#ifdef FPDFSDK_EXPORTS
+# if defined(_WIN32)
+#  define FPDF_EXPORT __declspec(dllexport)
+#  define FPDF_CALLCONV __stdcall
+# else
+#  define FPDF_EXPORT __attribute__((visibility("default")))
+#  define FPDF_CALLCONV
+# endif
 #else
-#define FPDF_EXPORT
-#define FPDF_CALLCONV
+# if defined(_WIN32)
+#  define FPDF_EXPORT __declspec(dllimport)
+#  define FPDF_CALLCONV __stdcall
+# else
+#  define FPDF_EXPORT
+#  define FPDF_CALLCONV
+# endif
 #endif
 
 // Exported Functions
