diff --git a/config/posix/BUILD.gn b/config/posix/BUILD.gn
index 66092908..4a7bef21 100644
--- a/config/posix/BUILD.gn
+++ b/config/posix/BUILD.gn
@@ -29,6 +29,10 @@ config("runtime_library") {
   lib_dirs = []
   libs = []
 
+  if(!is_clang && !use_custom_libcxx && libcpp_is_static) {
+    ldflags = [ "-static-libstdc++", "-static-libgcc" ]
+  }
+
   if (use_custom_libcxx) {
     if (libcpp_is_static) {
       # Don't leak any symbols on a static build.
