diff --git a/BUILD.gn b/BUILD.gn
index b4b93b584..d57bb555d 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -23,6 +23,7 @@ config("pdfium_common_config") {
     "OPJ_STATIC",
     "PNG_PREFIX",
     "PNG_USE_READ_MACROS",
+    "FPDFSDK_EXPORTS"
   ]
 
   if (pdf_use_skia) {
@@ -100,7 +101,7 @@ config("xfa_warnings") {
   }
 }
 
-static_library("pdfium") {
+shared_library("pdfium") {
   sources = [
     "fpdfsdk/cba_annotiterator.cpp",
     "fpdfsdk/cba_annotiterator.h",
