diff --git a/config/BUILDCONFIG.gn b/config/BUILDCONFIG.gn
index 26a23e459..b7517c4b2 100644
--- a/config/BUILDCONFIG.gn
+++ b/config/BUILDCONFIG.gn
@@ -480,6 +480,12 @@ if (is_fuchsia) {
 }
 set_defaults("shared_library") {
   configs = default_shared_library_configs
+  if (is_android) {
+    configs -= [ "//build/config/android:hide_all_but_jni_onload" ]
+    # By appending .cr, we prevent name collisions with libraries already
+    # loaded by the Android zygote. (Required for Android API 21-22)
+    output_extension = "cr.so"
+  }
 }
 set_defaults("loadable_module") {
   configs = default_shared_library_configs
