diff --git a/config/BUILDCONFIG.gn b/config/BUILDCONFIG.gn
index 105e194..50499f6 100644
--- a/config/BUILDCONFIG.gn
+++ b/config/BUILDCONFIG.gn
@@ -479,6 +479,9 @@ if (is_fuchsia) {
 }
 set_defaults("shared_library") {
   configs = default_shared_library_configs
+  if (is_android) {
+    configs -= [ "//build/config/android:hide_all_but_jni_onload" ]
+  }
 }
 set_defaults("loadable_module") {
   configs = default_shared_library_configs
