diff --git a/build_overrides/BUILDCONFIG.gn b/build_overrides/BUILDCONFIG.gn
index 768fe60e0..72c510471 100644
--- a/build_overrides/BUILDCONFIG.gn
+++ b/build_overrides/BUILDCONFIG.gn
@@ -475,6 +475,9 @@ if (is_android) {
 }
 set_defaults("shared_library") {
   configs = default_shared_library_configs
+  if (is_android) {
+    configs -= [ "//build/config/android:hide_all_but_jni_onload" ]
+  }
 }
 set_defaults("loadable_module") {
   configs = default_shared_library_configs
