diff --git a/config/BUILDCONFIG.gn b/config/BUILDCONFIG.gn
index b5dca4a24..4174baa9c 100644
--- a/config/BUILDCONFIG.gn
+++ b/config/BUILDCONFIG.gn
@@ -429,7 +429,6 @@ set_defaults("rust_library") {
 
 # Compute the set of configs common to all linked targets (shared libraries,
 # loadable modules, executables) to avoid duplication below.
-import("//build/config/cast.gni")
 if (is_win) {
   # Many targets remove these configs, so they are not contained within
   # //build/config:executable_config for easy removal.
@@ -446,13 +445,6 @@ if (is_win) {
   # Warn whenever linking code that could unintentionally switch off
   # branch target identification (BTI) on Arm systems.
   _linker_configs = [ "//build/config/android:lld_branch_target_hardening" ]
-} else if (is_linux && !enable_cast_receiver) {
-  # Warn whenever linking code that could unintentionally switch off
-  # branch target identification (BTI) on Arm systems.
-  #
-  # Note: BTI is disabled on cast builds, since cast is unlikely to have any
-  # PAC/BTI hardware.
-  _linker_configs = [ "//build/config/linux:lld_branch_target_hardening" ]
 } else {
   _linker_configs = []
 }
diff --git a/config/linux/BUILD.gn b/config/linux/BUILD.gn
index 3d5cf10cc..131bb71d1 100644
--- a/config/linux/BUILD.gn
+++ b/config/linux/BUILD.gn
@@ -6,9 +6,6 @@ import("//build/config/c++/c++.gni")
 import("//build/config/chromeos/ui_mode.gni")
 import("//build/config/linux/pkg_config.gni")
 import("//build/config/ui.gni")
-if (current_cpu == "arm64") {
-  import("//build/config/arm.gni")
-}
 
 group("linux") {
   visibility = [ "//:optimize_gn_gen" ]
@@ -19,17 +16,15 @@ group("linux") {
 # Linux-only. This is not applied to Android, but is applied to ChromeOS.
 config("compiler") {
   if (current_cpu == "arm64") {
+    import("//build/config/arm.gni")
     cflags = []
     asmflags = []
-    rustflags = [ "-Ctarget-feature=-outline-atomics" ]
     if (arm_control_flow_integrity == "standard") {
       cflags += [ "-mbranch-protection=standard" ]
       asmflags += [ "-mbranch-protection=standard" ]
-      rustflags += [ "-Zbranch-protection=pac-ret,bti" ]
     } else if (arm_control_flow_integrity == "pac") {
       cflags += [ "-mbranch-protection=pac-ret" ]
       asmflags += [ "-mbranch-protection=pac-ret" ]
-      rustflags += [ "-Zbranch-protection=pac-ret" ]
     }
   }
 }
@@ -73,18 +68,3 @@ if (use_glib) {
     ]
   }
 }
-
-config("lld_branch_target_hardening") {
-  # Config opts a shared library into BTI linker hardening. This
-  # is an opt-in config (rather than default-enabled) to avoid
-  # interfering with the V8 CFI bots (crbug.com/1334614).
-  if (current_cpu == "arm64") {
-    if (arm_control_flow_integrity == "standard") {
-      # Linking objects without GNU_PROPERTY_AARCH64_FEATURE_1_BTI
-      # in their .gnu.note section implicitly results in the final
-      # binary losing Branch Target Identification (BTI) support.
-      # Issue a warning if this happens.
-      ldflags = [ "-Wl,-z,force-bti" ]
-    }
-  }
-}
