diff --git a/toolchain/win/toolchain.gni b/toolchain/win/toolchain.gni
index 7dbd8a17e..361f74f85 100644
--- a/toolchain/win/toolchain.gni
+++ b/toolchain/win/toolchain.gni
@@ -252,7 +252,7 @@ template("msvc_toolchain") {
       } else {
         werror = ""
       }
-      command = "\"$python_path\" $_tool_wrapper_path rc-wrapper $env rc.exe /nologo $werror$sys_include_flags{{defines}} {{include_dirs}} /fo{{output}} {{source}}"
+      command = "rc.exe /nologo $werror$sys_include_flags{{defines}} {{include_dirs}} /fo{{output}} /dENV=$env {{source}}"
       depsformat = "msvc"
       outputs = [ "$object_subdir/{{source_name_part}}.res" ]
       description = "RC {{output}}"
