diff --git a/toolchain/win/setup_toolchain.py b/toolchain/win/setup_toolchain.py
index 68ad6c4a0..210900ad8 100644
--- a/toolchain/win/setup_toolchain.py
+++ b/toolchain/win/setup_toolchain.py
@@ -21,7 +21,7 @@ sys.path.append(os.path.join(os.path.dirname(__file__), os.pardir, os.pardir))
 import gn_helpers
 
 SCRIPT_DIR = os.path.dirname(__file__)
-SDK_VERSION = '10.0.28000.0'
+SDK_VERSION = os.environ.get('WINDOWS_SDK_VERSION', '10.0.28000.0')
 MSVC_DIR = re.compile('^.*/VC/Tools/MSVC/[^/]+/include$')
 WINDOWS_KITS_DIR = re.compile(r'^(.*/Windows Kits/\d+/Include/[^/]+)/.*')
 
diff --git a/toolchain/win/toolchain.gni b/toolchain/win/toolchain.gni
index 9a7e27ca7..0d29156a1 100644
--- a/toolchain/win/toolchain.gni
+++ b/toolchain/win/toolchain.gni
@@ -270,7 +270,7 @@ template("msvc_toolchain") {
       } else {
         werror = ""
       }
-      command = "\"$python_path\" $_tool_wrapper_path rc-wrapper $env rc.exe /nologo $werror$sys_include_flags{{defines}} {{include_dirs}} /fo{{output}} {{source}}"
+      command = "rc.exe /nologo $werror$sys_include_flags{{defines}} {{include_dirs}} /fo{{output}} /dENV=$env {{source}}"
       depsformat = "msvc"
       outputs = [ "$object_subdir/{{source_name_part}}.res" ]
       description = "RC {{output}}"
