diff --git a/core/fxge/BUILD.gn b/core/fxge/BUILD.gn
index a4deadb1c..12994a22a 100644
--- a/core/fxge/BUILD.gn
+++ b/core/fxge/BUILD.gn
@@ -167,7 +167,7 @@ source_set("fxge") {
     sources += [ "linux/fx_linux_impl.cpp" ]
   }
 
-  if (is_mac) {
+  if (is_mac || is_ios) {
     sources += [
       "apple/fx_apple_impl.cpp",
       "apple/fx_apple_platform.cpp",
diff --git a/testing/test.gni b/testing/test.gni
index 9d528da98..2c2a6706f 100644
--- a/testing/test.gni
+++ b/testing/test.gni
@@ -207,11 +207,6 @@ template("test") {
       }
 
       _bundle_id_suffix = target_name
-      if (ios_automatically_manage_certs) {
-        # Use the same bundle identifier for all unit tests when managing
-        # certificates automatically as the number of free certs is limited.
-        _bundle_id_suffix = "generic-unit-test"
-      }
       if (!defined(extra_substitutions)) {
         extra_substitutions = []
       }
