diff --git a/BUILD.gn b/BUILD.gn
index 81d944f05e9..c64c2e9ce7f 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -5089,7 +5089,7 @@ v8_header_set("v8_internal_headers") {
 
   # BrowserEngineKit only exists on iOS itself, not on tvOS and other
   # iOS-derived platforms, so we need to check `target_platform` here.
-  if (is_ios && target_platform == "iphoneos") {
+  if (is_ios && target_platform == "iphoneos" && target_environment != "catalyst") {
     frameworks += [ "BrowserEngineKit.framework" ]
   }
 
diff --git a/src/base/build_config.h b/src/base/build_config.h
index ee745539dfd..188f8d8c6de 100644
--- a/src/base/build_config.h
+++ b/src/base/build_config.h
@@ -45,6 +45,7 @@
 
 // BrowserEngineCore JIT write protect is only available on iOS 17.4 and later.
 #if defined(V8_HOST_ARCH_ARM64) && defined(V8_OS_IOS) && \
+    !defined(TARGET_OS_MACCATALYST) && \
     defined(__IPHONE_17_4) &&                            \
     __IPHONE_OS_VERSION_MIN_REQUIRED >= __IPHONE_17_4
 #define V8_HAS_BECORE_JIT_WRITE_PROTECT 1
