diff --git a/config/BUILDCONFIG.gn b/config/BUILDCONFIG.gn
index 1ed842b77..b41d3e293 100644
--- a/config/BUILDCONFIG.gn
+++ b/config/BUILDCONFIG.gn
@@ -135,6 +135,9 @@ declare_args() {
   # set "is_official_build" to true for any build intended to ship to end-users.
   is_official_build = false
 
+  # Use musl instead of glibc
+  is_musl = false
+
   # Set to true when compiling with the Clang compiler.
   is_clang = current_os != "linux" ||
              (current_cpu != "s390x" && current_cpu != "s390" &&
@@ -237,7 +240,9 @@ if (target_os == "android") {
   _default_toolchain = "//build/toolchain/android:android_clang_$target_cpu"
 } else if (target_os == "chromeos" || target_os == "linux") {
   # See comments in build/toolchain/cros/BUILD.gn about board compiles.
-  if (is_clang) {
+  if (is_musl) {
+    _default_toolchain = "//build/toolchain/linux/musl:$target_cpu"
+  } else if (is_clang) {
     _default_toolchain = "//build/toolchain/linux:clang_$target_cpu"
   } else {
     _default_toolchain = "//build/toolchain/linux:$target_cpu"
diff --git a/config/c++/c++.gni b/config/c++/c++.gni
index bd8e7118e..87d38bb12 100644
--- a/config/c++/c++.gni
+++ b/config/c++/c++.gni
@@ -61,7 +61,7 @@ use_safe_libcxx = (use_custom_libcxx && enable_safe_libcxx) || is_nacl_saigo
 
 # libstdc++ has its own hardening assertions that we want to enable by default
 # in Chromium builds.
-use_safe_libstdcxx = is_linux && !use_custom_libcxx && enable_safe_libstdcxx
+use_safe_libstdcxx = is_linux && !use_custom_libcxx
 
 # libc++abi needs to be exported from executables to be picked up by shared
 # libraries on certain instrumented builds.
diff --git a/config/compiler/BUILD.gn b/config/compiler/BUILD.gn
index 24e059596..ba91b474c 100644
--- a/config/compiler/BUILD.gn
+++ b/config/compiler/BUILD.gn
@@ -379,7 +379,7 @@ config("compiler") {
         } else {
           cflags += [ "-fstack-protector" ]
         }
-      } else if ((is_posix && !is_chromeos && !is_nacl) || is_fuchsia) {
+      } else if ((is_posix && !is_chromeos && !is_nacl && !is_musl) || is_fuchsia) {
         if (current_os != "aix") {
           # Not available on aix.
           cflags += [ "-fstack-protector" ]
