diff --git a/crc32_simd.c b/crc32_simd.c
index cbe9739..4d01b1b 100644
--- a/crc32_simd.c
+++ b/crc32_simd.c
@@ -398,7 +398,7 @@ uint32_t ZLIB_INTERNAL crc32_sse42_simd_(  /* SSE4.2+PCLMUL */
  */
 #include <arm_acle.h>
 #include <arm_neon.h>
-#define TARGET_ARMV8_WITH_CRC __attribute__((target("arch=armv8-a+crc")))
+#define TARGET_ARMV8_WITH_CRC
 #else  // !defined(__GNUC__) && !defined(_aarch64__)
 #error ARM CRC32 SIMD extensions only supported for Clang and GCC
 #endif
