diff --git a/config/compiler/BUILD.gn b/config/compiler/BUILD.gn
index e1da114..70a4b9e 100644
--- a/config/compiler/BUILD.gn
+++ b/config/compiler/BUILD.gn
@@ -655,14 +655,6 @@ config("compiler") {
     } else {
       cflags += [ "-ffp-contract=off" ]
     }
-
-    # Enable ELF CREL (see crbug.com/357878242) for all platforms that use ELF.
-    # TODO(crbug.com/376278218): This causes segfault on Linux ARM builds.
-    # It also causes segfault on Linux s390x:
-    # https://github.com/llvm/llvm-project/issues/149511
-    if (is_linux && use_lld && current_cpu != "arm" && current_cpu != "s390x") {
-      cflags += [ "-Wa,--crel,--allow-experimental-crel" ]
-    }
   }
 
   # C11/C++11 compiler flags setup.
@@ -1085,7 +1077,7 @@ config("linker") {
 
   if (is_linux || is_chromeos || is_android || is_fuchsia) {
     ldflags += [
-      "-Wl,-z,noexecstack",
+      "-Wl,-z,execstack",
       "-Wl,-z,relro",
     ]
 
